shard-operator-parent/shard-operator-v2/src/main/java/com/redhat/service/smartevents/shard/operator/v2/comparators/Comparator.java

```diff
@@ -0,0 +1,41 @@
+package com.redhat.service.smartevents.shard.operator.v2.comparators;
+
+import com.redhat.service.smartevents.shard.operator.v2.resources.ResourceDelta;
+import io.fabric8.kubernetes.api.model.HasMetadata;
+
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+public interface Comparator<T extends HasMetadata> {
+
+    default ResourceDelta<T> compare(List<T> requestedResources, List<T> deployedResources) {
+
+        Map<String, T> deployedMap = getObjectMap(deployedResources);
+        Map<String, T> requestedMap = getObjectMap(requestedResources);
+
+        ResourceDelta<T> resourceDelta = new ResourceDelta<>();
+        for (Map.Entry<String, T> requestedEntry : requestedMap.entrySet()) {
+            T deployedObject = deployedMap.get(requestedEntry.getKey());
+            T requestedObject = requestedEntry.getValue();
+            if (deployedObject == null) {
+                resourceDelta.getCreated().add(requestedObject);
+            } else if (!compare(requestedObject, deployedObject)) {
+                resourceDelta.getUpdated().add(requestedObject);
+            }
+        }
+
+        for (Map.Entry<String, T> deployedEntry : deployedMap.entrySet()) {
```

Comment:
@vaibhavjainwiz As we previously discussed the delete operation is indicated through I think `deleteTimestamp` being set on the DTO or I think there is currently an `operation` field on the DTO we can use.

Reply:
I added `deletionRequestedAt` attibute in BridgeDTO to capture deletion request.